psh/test-control.py

```diff
@@ -0,0 +1,91 @@
+# Phoenix-RTOS
+#
+# phoenix-rtos-tests
+#
+# psh control codes test
+#
+# Copyright 2023 Phoenix Systems
+# Author: Damian Modzelewski
+#
+# This file is part of Phoenix-RTOS.
+#
+# %LICENSE%
+#
+
+import tools.psh as psh
+
+
+def CC_parser(harness, input: str, output: str):
+    harness.write(input)
+    if not output:
+        harness.expect(psh.PROMPT)
+    else:
+        harness.expect(f"psh: {output} not found", timeout=-1)
+
+
+"""
+CONTROL CODE CHEAT-SHEAT
+
+(\177 or \033[3~) -> Delete the last character
+\001 -> Move to the first position in the text
+\003 -> Cancel command
+\003[A -> In system declaration of Up arrow or 8 on numeric keypad
+(\004 or \033[3~) -> Delete the next character after the cursor
+\005 -> Move to the last possible position
+(\002 or \033[D) -> Move the cursor to the left
+(\006 or \033[C) -> Move the cursor to the right
+\013 -> Cut text between cursor and end of line
+\033[B -> In system declaration of Down arrow or 2 on numeric keypad
+(\031 or \033[2~) -> Paste what is in the clipboard. The second one uses 0 on the numeric keypad.
+\027 -> Cut the last phrase before the cursor
+\025 -> Cut text between the beginning of the line and the cursor
+\033[1;5D -> Combination of keys CTRL + 4 on numeric keypad or left arrow to move the cursor to the left over phrases
+\033[1;5C -> Combination of keys CTRL + 6 on numeric keypad or right arrow to move the cursor to the left over phrases
+\t -> autocomplete path
+
+"""
+
+
+@psh.run
+def harness(p):
```

Comment:
I'm considering whether to organize the tests in a uniform manner, for example, in relation to the CONTROL CODE CHEAT SHEET, and whether to expand the tests from basic ones that test a given control code, to tests that use several, and up to those that use as many as possible. It would be beneficial if the combinations reflected real-world scenarios. So, then you could also  divide the tests into separate functions (?)

Reply:
This test is just to check that this control codes works, doesn't need to test them with full possibility use set.